### ISSUE TITLE
Add HSTS support to GitHub Pages custom domains.

### ISSUE THREAD

**Issue by bardiharborow** [isaacs/github#1249]
> Hey GitHub,
>
> A big thanks for launching HTTPS support on GitHub Pages for custom domains. The final loose-end stopping many people from migrating is the lack of a option to serve a HSTS header (and possibly other related security headers). If you could add this to the feature request list, that would be appreciated.
>
> Regards,
> Bardi Harborow

**Comment by Geoegroeg** [comment]
Still no HSTS support. Especially inconvenient for Cloudflare users who have other (sub)domains on the domain used for GitHub Pages, universally unable to activate HSTS because it will break GitHub Pages (no origin certificate) and unable to activate Full (strict) SSL via Cloudflare.

**Comment by MatTheCat** [comment]
Is it on the roadmap?

**Comment by brandonseydel** [comment]
This feature needs to be done ASAP.

**Comment by nyghtly-derek** [comment]
Would like to see this feature implemented so that I can bump my Mozilla Observatory scores: https://observatory.mozilla.org/analyze/nyghtly.me

**Comment by pistell** [comment]
It’s been almost 2 years and no response 😟

**Comment by wangqr** [comment]
This has been a urgent need as we are seeing massive MitM for github.io sites

**Comment by strugee** [comment]
@TPS I think adding the `vulnerability alerts` label to this issue was probably a mistake?

**Comment by anthonyrom** [comment]
Github please do this

**Comment by haliphax** [comment]
Just realized this was a thing when I ran a webpagetest.org scan against my GitHub Pages-hosted site. Eek! Throwing my upvote in here.

**Comment by jplatte** [comment]
> HSTS seems to work automatically on Pages sites created by users who created their accounts after the afternoon of June 15, 2016. The key here is when an account is created, not the pages site itself.
> 
> The good news is that a pages site created by an organisation created after that date will us HSTS. I’ve checked it out with my own pre-2016 account (no) and a brand new organisation account owned by my account (yes).
> 
> So your personal username.github.io can’t have it just yet (we’re trying to not break old pages sites for now) but if you create a new organisation for a pages site, you’ll have it there.
>
> — https://github.community/t/feature-request-hsts-for-github-pages/176265

**Comment by haliphax** [comment]
Consider me thoroughly confused. I went to the GitHub Feedback site and saw no results when searching for "HSTS". Wasn't aware there was a _separate_ "community" site where suggestions are supposed to go. At any rate, there's mention of it in both now, as I started a feedback post. :laughing: https://github.com/github/feedback/discussions/4444

**Comment by Levi-Lesches** [comment]
Correct, the new repo is https://github.com/github/feedback/, and you can open a discussion there in the [general feedback](https://github.com/github/feedback/discussions/categories/general-feedback) discussions group. This is a fairly recent change, and it only just made it to this repo's README yesterday or something. This repo (and I believe github.community) are obsolete now.

**Comment by haliphax** [comment]
Cool. Guess it's worth having something in the new feedback bucket, then.

**Comment by lzcapp** [comment]
And it has been 3 years...😅  I am even considering moving my sites out!

**Comment by cloewen8** [comment]
Pretty annoyed as well (although I can only be so annoyed, it is an extra feature for a cheap service).
No mention of this feedback discussion, no updates here (by staff, for 3 years) and no way to even opt into HSTS (new sites gain access, why not old, but compatible, sites).
